RFS/src/main/java/com/rfs/version_universal/RemoteIndexMetadataFactory.java

```diff
@@ -0,0 +1,52 @@
+package com.rfs.version_universal;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.dataformat.smile.SmileFactory;
+
+import com.rfs.common.SnapshotRepo.Provider;
+import com.rfs.models.IndexMetadata;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@RequiredArgsConstructor
+@Slf4j
+public class RemoteIndexMetadataFactory implements IndexMetadata.Factory {
+
+    private final RemoteReaderClient client;
+
+    private ObjectNode indexData;
+
+    ObjectNode getIndexData() {
+        if (indexData == null) {
+            indexData = client.getIndexes();
+        }
+        return indexData;
+    }
+
+    @Override
+    public IndexMetadata fromRepo(String snapshotName, String indexName) {
+        log.info("Using remote cluster directly");
+        return new RemoteIndexMetadata(indexName, (ObjectNode)getIndexData().get(indexName));
+    }
+
+    @Override
+    public Provider getRepoDataProvider() {
+        return new RemoteSnapshotDataProvider(getIndexData());
+    }
+
+    @Override
+    public IndexMetadata fromJsonNode(JsonNode root, String indexId, String indexName) {
+        throw new UnsupportedOperationException("Unimplemented method 'fromJsonNode'");
```

Comment:
I'm not sure what the "correct" abstraction here is, but the other parts of the interfaces seem much improved in their coherence and I'm willing to let this particular mismatch exist for the moment.  We seem to be headed in the right direction even if we're not all the way there yet.

Reply:
Yup - I'll be paying attention to opportunities in this space as we continue iterating.